src/System.Management.Automation/FormatAndOutput/common/PSStyle.cs

```diff
@@ -348,29 +413,141 @@ public sealed class FileInfoFormatting
+                }
+
+                /// <summary>
+                /// Remove an extension from dictionary.
+                /// </summary>
+                /// <param name="extension">Extension to remove.</param>
+                public void Remove(string extension)
+                {
+                    _extensionDictionary.Remove(ValidateExtension(extension));
+                }
+
+                /// <summary>
+                /// Clear the dictionary.
+                /// </summary>
+                public void Clear()
+                {
+                    _extensionDictionary.Clear();
+                }
+
+                /// <summary>
+                /// Gets the enumerator for the dictionary.
+                /// </summary>
+                /// <returns>The enumerator for the dictionary.</returns>
+                public IEnumerator GetEnumerator()
+                {
+                    return _extensionDictionary.GetEnumerator();
+                }
+
+                /// <summary>
+                /// Gets or sets the decoration by specified extension.
+                /// </summary>
+                /// <param name="extension">Extension to get decoration for.</param>
+                /// <returns>The decoration for specified extension.</returns>
+                public string this[string extension]
+                {
+                    get
+                    {
+                        return _extensionDictionary[ValidateExtension(extension)];
+                    }
+
+                    set
+                    {
+                        _extensionDictionary[ValidateExtension(extension)] = ValidateNoContent(value);
+                    }
+                }
+
+                /// <summary>
+                /// Gets whether the dictionary contains the specified extension.
+                /// </summary>
+                /// <param name="extension">Extension to check for.</param>
+                /// <returns>True if the dictionary contains the specified extension, otherwise false.</returns>
+                public bool ContainsKey(string extension)
+                {
+                    return _extensionDictionary.ContainsKey(ValidateExtension(extension));
+                }
+
+                /// <summary>
+                /// Gets the extensions for the dictionary.
+                /// </summary>
+                /// <returns>The extensions for the dictionary.</returns>
+                public IEnumerable<string> Keys
+                {
+                    get
+                    {
+                        return _extensionDictionary.Keys;
+                    }
+                }
+            }
 
             /// <summary>
             /// Gets the style for archive.
             /// </summary>
-            public Dictionary<string, string> Extension { get; }
+            public FileExtensionDictionary Extension { get; }
```

Comment:
Wait if this is a public property, changing it from a `Dictionary<string, string>` to something that doesn't implement that is a breaking change. If this feature is still in preview, we can live with that, but we should take a close look at the actual type being exported to ensure it's fit to be public.

Reply:
This is ExperimentalFeature so this is not a breaking change